### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Abelian): AB5 implies AB4

### PR DESCRIPTION
**Pull request by javra**
An AB4 category is a category in which taking coproducts is exact, an AB5 category is one where taking filtered colimits is exact. This shows that any Abelian catgory which is AB5 is also AB4.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/Preserves/FunctorCategory.lean [hunk 1]
```diff
@@ -71,15 +73,25 @@ def FunctorCategory.prodPreservesColimits [HasBinaryProducts D] [HasColimits D]
...
+    [HasLimitsOfShape J D] (F : C ⥤ E) :
+    PreservesLimitsOfShape J ((whiskeringLeft C E D).obj F) :=
+  ⟨fun {K} =>
+    ⟨fun c {hc} => by
+      apply evaluationJointlyReflectsLimits
+      intro Y
+      change IsLimit (((evaluation E D).obj (F.obj Y)).mapCone c)
+      exact PreservesLimit.preserves hc⟩⟩
+
+instance whiskeringLeftPreservesLimits [HasLimitsOfSize.{(max u₁ v₂)} D] (F : C ⥤ E) :
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Limits/Preserves/FunctorCategory.lean:92]: Could you add a more general `PreservesLimitsOfSize` instance, with arbitrary universe parameters and suitable `HasLimitsOfSize` assumption?

Comment (javra) [reply Mathlib/CategoryTheory/Limits/Preserves/FunctorCategory.lean]: We don't need the more specific one then, do we? Mirrored the way `whiskeringRightPreservesLimits` does it.

File: Mathlib/CategoryTheory/Abelian/GrothendieckAxioms.lean [hunk 2]
```diff
@@ -90,4 +95,32 @@ class AB5Star [HasCofilteredLimits C] where
...
+    letI : PreservesFiniteLimits ((whiskeringLeft (Discrete I) (Discrete α) C).obj
+        (Discrete.functor fun x ↦ ↑x)) :=
+      ⟨fun J _ _ => whiskeringLeftPreservesLimitsOfShape J _⟩
+    letI : PreservesFiniteLimits ((whiskeringLeft (Discrete I) (Discrete α) C).obj
+        (Discrete.functor (·.val)) ⋙ colim) :=
+      compPreservesFiniteLimits _ _
+    preservesFiniteLimitsOfNatIso (liftToFinsetEvaluationIso  I).symm
+
+/-- A category with finite biproducts and finite limits is AB4 if it is AB5. -/
+def AB4.ofAB5 [HasCoproducts C] [HasFilteredColimits C] [AB5 C] : AB4 C where
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Abelian/GrothendieckAxioms.lean:118]: A `HasFiniteCoproducts` assumption would be more consistent with the docstring, and it seems we have `hasCoproducts_of_finite_and_filtered`.

**Comment by joelriou** [comment]
Could you edit the first message above in order to add a PR description, where you would remind the reader what AB4 and AB5 are.

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**